backend/otel/syslog_test.go

```diff
@@ -1,16 +1,28 @@
 package otel
 
 import (
-	"github.com/stretchr/testify/assert"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 func Test_extractSyslog(t *testing.T) {
-	fields := extractedFields{
-		logBody: "<1>1 2023-07-27T05:43:22.401882Z render render-log-endpoint-test 1 render-log-endpoint-test - Render test log",
-	}
-	attrs := map[string]any{}
-	extractSyslog(&fields, attrs)
-	assert.Equal(t, "render", attrs["hostname"])
+	fields := newExtractedFields()
+
+	fields.logBody = "<1>1 2023-07-27T05:43:22.401882Z render render-log-endpoint-test 1 render-log-endpoint-test - Render test log"
+	extractSyslog(fields)
+	assert.Equal(t, "render", fields.attrs["hostname"])
 	assert.Equal(t, "Render test log", fields.logBody)
 }
+
+func Test_extractSyslogWithStructuredData(t *testing.T) {
```

Comment:
This is unrelated but our syslog parsing for structured data wasn't extracting data correctly. Partly because render.com doesn't support this and partly because our typing wasn't as string to catch this before.